### PR TITLE
[10.0][FIX] medical_center and medical_physician

### PR DESCRIPTION
**Pull request by woodbrettm**
* Fix parent_id field in medical_center
* Add primary_center and secondary_centers to medical_physician

### CODE REVIEW

File: medical_physician/models/medical_physician.py [hunk 1]
```diff
@@ -38,6 +38,17 @@ class MedicalPhysician(models.Model):
         comodel_name='medical.physician.schedule.template',
         inverse_name='physician_id',
     )
+    medical_center_primary_id = fields.Many2one(
```
Comment (lasley) [inline medical_physician/models/medical_physician.py]: This should not be renamed

File: medical_physician/views/medical_physician_view.xml [hunk 2]
```diff
@@ -16,7 +16,7 @@
             <xpath expr="//field[@name='email']" position="after">
                 <field name="code" />
                 <field name="specialty_id" />
-                <field name="parent_id" />
+                <field name="medical_center_primary_id" />
```
Comment (lasley) [inline medical_physician/views/medical_physician_view.xml]: Why is parent_id removed?

File: medical_physician/views/medical_physician_view.xml [hunk 3]
```diff
@@ -33,10 +33,12 @@
             </xpath>
             <xpath expr="//h1" position="after">
                 <h3>
-                    <field name="parent_id"
-                           placeholder="Medical Center"
-                           domain="[('type', '=', 'medical.center')]"
-                           context="{'default_type': 'medical.center'}"
+                    <field name="medical_center_primary_id"
```
Comment (lasley) [inline medical_physician/views/medical_physician_view.xml]: Why is parent_id removed?

**Review by lasley — CHANGES_REQUESTED** [review]

File: medical_physician/models/medical_physician.py (hunk 1 above)
Comment (woodbrettm) [reply medical_physician/models/medical_physician.py]: I'm not sure I follow. Renamed from what?

Comment (lasley) [reply medical_physician/models/medical_physician.py]: The existing medical center concept was not supposed to be changed. I believe it used to be parent_id

Comment (woodbrettm) [reply medical_physician/models/medical_physician.py]: ah I see. So I guess this should be removed, and then just have something like `secondary_parent_ids` for the other field?

Comment (lasley) [reply medical_physician/models/medical_physician.py]: `medical_center_ids`, compute and store. Make computation dependent on `parent_id`. Ensure that the `parent_id`'s medical center record is added in when `parent_id` is changed. Keep in mind, parent will be a partner so it lacks the medical context.

Comment (woodbrettm) [reply medical_physician/models/medical_physician.py]: @lasley is the compute function only to add in the parent_id to medical_center_ids? Or is there something else that needs to be added to medical_center_ids as well?

Should I also add an inverse function to the field?

Comment (lasley) [reply medical_physician/models/medical_physician.py]: Yes it is just to add the parent_id into medical_center_ids.

Inverse is not needed if it's stored (I think).

Comment (woodbrettm) [reply medical_physician/models/medical_physician.py]: @lasley you want the field to be editable though right? Or just readonly?

Comment (lasley) [reply medical_physician/models/medical_physician.py]: It needs to be editable

File: medical_physician/models/medical_physician.py [hunk 4]
```diff
@@ -38,6 +38,28 @@ class MedicalPhysician(models.Model):
...
+        for record in self:
+            center = self.env['medical.center'].search([
+                ('partner_id', '=', record.parent_id.id),
+            ], limit=1)
+
+            if not center:
+                continue
+
+            if center not in record.medical_center_ids:
+                record.medical_center_ids += center
```
Comment (woodbrettm) [inline medical_physician/models/medical_physician.py]: So I added a compute method for `medical_center_ids` with this section here. What's really weird is whenever I change `parent_id` in the UI, `medical_center_ids` is completely emptied and this single center record is added. Any ideas as to why that may be?

Comment (woodbrettm) [reply medical_physician/models/medical_physician.py]: Oh actually no I think this bit is working. However I think what happens is when the compute function is triggered, the field is completely emptied of the records before running code in the compute method. I'm not exactly sure what to do in that case.

Comment (lasley) [reply medical_physician/models/medical_physician.py]: Why aren't you using the ORM methods?

Comment (lasley) [reply medical_physician/models/medical_physician.py]: `[(4, center.id)]`

Comment (woodbrettm) [reply medical_physician/models/medical_physician.py]: I tried that but get the following error:
```
2017-05-16 23:35:29,118 24933 INFO physician odoo.sql_db: bad query:  INSERT INTO medical_center_medical_physician_rel (medical_physician_id, medical_center_id)
                        (SELECT a, b FROM unnest('{}') AS a, unnest(ARRAY[1]) AS b)
                        EXCEPT (SELECT medical_physician_id, medical_center_id FROM medical_center_medical_physician_rel WHERE medical_physician_id IN ())

2017-05-16 23:35:29,118 24933 ERROR physician odoo.http: Exception during JSON request handling.
Traceback (most recent call last):
  File "/opt/odoo/odoo/http.py", line 638, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/opt/odoo/odoo/http.py", line 675, in dispatch
    result = self._call_function(**self.params)
  File "/opt/odoo/odoo/http.py", line 331, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/opt/odoo/odoo/service/model.py", line 119, in wrapper
    return f(dbname, *args, **kwargs)
  File "/opt/odoo/odoo/http.py", line 324, in checked_call
    result = self.endpoint(*a, **kw)
  File "/opt/odoo/odoo/http.py", line 933, in __call__
    return self.method(*args, **kw)
  File "/opt/odoo/odoo/http.py", line 504, in response_wrap
    response = f(*args, **kw)
  File "/opt/odoo/addons/web/controllers/main.py", line 862, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/opt/odoo/addons/web/controllers/main.py", line 854, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/opt/odoo/odoo/api.py", line 681, in call_kw
    return call_kw_multi(method, model, args, kwargs)
  File "/opt/odoo/odoo/api.py", line 672, in call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/opt/odoo/odoo/models.py", line 5469, in onchange
    newval = record[name]
  File "/opt/odoo/odoo/models.py", line 5177, in __getitem__
    return self._fields[key].__get__(self, type(self))
  File "/opt/odoo/odoo/fields.py", line 870, in __get__
    self.determine_draft_value(record)
  File "/opt/odoo/odoo/fields.py", line 988, in determine_draft_value
    self._compute_value(record)
  File "/opt/odoo/odoo/fields.py", line 926, in _compute_value
    getattr(records, self.compute)()
  File "/media/sf_odoo10dev/vertical-medical/medical_physician/models/medical_physician.py", line 63, in _compute_medical_center_ids
    'medical_center_ids': [(4, center.id)],
  File "/opt/odoo/addons/mail/models/mail_thread.py", line 275, in write
    result = super(MailThread, self).write(values)
  File "/opt/odoo/odoo/models.py", line 3545, in write
    self._write(old_vals)
  File "/opt/odoo/odoo/models.py", line 3661, in _write
    field.write(self.with_context(rel_context), vals[name])
  File "/opt/odoo/odoo/fields.py", line 2376, in write
    link([act[1]])
  File "/opt/odoo/odoo/fields.py", line 2346, in link
    cr.execute(query, (records.ids, list(sub_ids), tuple(records.ids)))
  File "/opt/odoo/odoo/sql_db.py", line 141, in wrapper
    return f(self, *args, **kwargs)
  File "/opt/odoo/odoo/sql_db.py", line 218, in execute
    res = self._obj.execute(query, params)
ProgrammingError: syntax error at or near ")"
LINE 3: ...nter_medical_physician_rel WHERE medical_physician_id IN ())
```

The code I have is as follows:
```python
    @api.multi
    @api.depends('parent_id')
    def _compute_medical_center_ids(self):
        for record in self:
            center = self.env['medical.center'].search([
                ('partner_id', '=', record.parent_id.id),
            ], limit=1)

            if not center:
                continue

            if center not in record.medical_center_ids:
                record.write({
                    'medical_center_ids': [(4, center.id)],
                })
```

Comment (lasley) [reply medical_physician/models/medical_physician.py]: Try uninstalling mail_thread_medical_prescription

Comment (woodbrettm) [reply medical_physician/models/medical_physician.py]: I already had that module uninstalled sadly :(

Comment (lasley) [reply medical_physician/models/medical_physician.py]: `medical.physician` needs to inherit from `mail.thread`. But this should already be the case by its inheritance of the [abstract.entity](https://github.com/LasLabs/vertical-medical/blob/release/10.0/medical/models/medical_abstract_entity.py#L15). Try directly inheriting as a test

Comment (lasley) [reply medical_physician/models/medical_physician.py]: Wait actually that's not it. When are you getting this error?

Comment (woodbrettm) [reply medical_physician/models/medical_physician.py]: I get that error whenever I change the parent_id in the physician form view. Adding centers to the medical_center_ids field works in the form though.

**Comment by woodbrettm** [comment]
@lasley removed physician changes. Good to go.

**Review by lasley — APPROVED** [review]

**Pull request merged**